splink/comparison_library.py

```diff
@@ -296,3 +296,79 @@ def jaro_winkler_at_thresholds(
         m_probability_or_probabilities_lev,
         m_probability_else,
     )
+
+
+class ArrayIntersectAtSizesComparisonBase(Comparison):
+    def __init__(
+        self,
+        col_name: str,
+        size_or_sizes: Union[int, list] = [1],
+        m_probability_or_probabilities_sizes: Union[float, list] = None,
+        m_probability_else=None,
+    ) -> Comparison:
+        """A comparison of the data in array column `col_name` with various
+        intersection sizes to assess similarity levels.
+
+        An example of the output with default arguments and setting
+        `size_or_sizes = [3, 1]` would be
+        - Intersection has at least 3 elements
+        - Intersection has at least 1 element (i.e. 1 or 2)
+        - Anything else (i.e. empty intersection)
+
+        Args:
+            col_name (str): The name of the column to compare
+            size_or_sizes (Union[int, list], optional): The size(s) of intersection
+                to use for the non-'else' similarity level(s). Should be in
+                descending order. Defaults to [1].
+            m_probability_or_probabilities_sizes (Union[float, list], optional):
+                _description_. If provided, overrides the default m probabilities
+                for the sizes specified. Defaults to None.
+            m_probability_else (_type_, optional): If provided, overrides the
+                default m probability for the 'anything else' level. Defaults to None.
+
+        Returns:
+            Comparison:
+        """
+
+        sizes = ensure_is_iterable(size_or_sizes)
+        if len(sizes) == 0:
```

Comment:
This should probably be:
```suggestion
        if len(sizes) <= 0:
```

So that users can't enter negative values.

Reply:
(see below)